pkg/morph/deploy/deploy.go

```diff
@@ -0,0 +1,205 @@
+// Package deploy provides NeoFS Sidechain deployment functionality.
+package deploy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/nspcc-dev/neo-go/pkg/core/block"
+	"github.com/nspcc-dev/neo-go/pkg/core/state"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neo-go/pkg/neorpc"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/actor"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract/manifest"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract/nef"
+	"github.com/nspcc-dev/neo-go/pkg/wallet"
+	"go.uber.org/zap"
+)
+
+// Blockchain groups services provided by particular Neo blockchain network
+// representing NeoFS Sidechain that are required for its deployment.
+type Blockchain interface {
+	// RPCActor groups functions needed to compose and send transactions to the
+	// blockchain.
+	actor.RPCActor
+
+	// GetCommittee returns list of public keys owned by Neo blockchain committee
+	// members. Resulting list is non-empty, unique and unsorted.
+	GetCommittee() (keys.PublicKeys, error)
+
+	// GetContractStateByID returns network state of the smart contract by its ID.
+	// GetContractStateByID returns error with 'Unknown contract' substring if
+	// requested contract is missing.
+	GetContractStateByID(id int32) (*state.Contract, error)
+
+	// ReceiveBlocks starts background process that forwards new blocks of the
+	// blockchain to the provided channel. The process handles all new blocks when
+	// ReceiveBlocks is called with nil filter. Returns unique identifier to be used
+	// to stop the process via Unsubscribe.
+	ReceiveBlocks(*neorpc.BlockFilter, chan<- *block.Block) (id string, err error)
+
+	// Unsubscribe stops background process started by ReceiveBlocks by ID.
+	Unsubscribe(id string) error
+}
+
+// KeyStorage represents storage of the private keys.
+type KeyStorage interface {
```

Comment:
why is it an interface? you have literally described its (the only) implementation in a few lines

Reply:
cuz package doesn't care how exactly key is stored (in file, in db, in memory, remotely)

Reply:
why do we have to store it? i understand your point but it is not about storing only but about generating too. why?

Reply:
we have to store it because we must not loose it between stages (which is really possible). This also relates to https://github.com/neo-project/proposals/issues/161 that will simplify this stuff with distributed single private key

Reply:
> we have to store it because we must not loose it between stages

do you mean that initialization should be idempotent? if no, can just a local variable save us?

Reply:
yep, as much as possible ofc. File can dissapear and there is nothing else we could do with this. But in 97% cases this will work

Reply:
> there is nothing else we can do but init new key

is it a problem? if no key has persisted, i would just init the next one? but if a key has persisted successfully, is it possible to just read it? mb i got the idea wrong

@roman-khimov

Reply:
If the key is on the chain, that's where the key is. If it's not, it doesn't exist and we don't care which one is gonna be used for the next attempt. The only problematic case probably is if you have an in-flight transaction that sets the key, but you restart before it's processed. This can be solved by always waiting for a new block initially (so that it'll bring this transaction with it), when the leader goes into this routine. It's not precisely accurate, but a sufficiently good way.